File: server/paper/src/main/java/su/plo/voice/paper/PaperVoiceLoader.java
package su.plo.voice.paper;

import org.bukkit.plugin.java.JavaPlugin;

public final class PaperVoiceLoader extends JavaPlugin {

    private final PaperVoiceServer voiceServer = new PaperVoiceServer(this);

    @Override
    public void onLoad() {
        voiceServer.onLoad();
    }

    @Override
    public void onEnable() {
        voiceServer.onInitialize();
    }

    @Override
    public void onDisable() {
        voiceServer.onShutdown();
    }
}

File: server/paper/src/main/kotlin/su/plo/voice/paper/PaperVoiceLoader.kt
package su.plo.voice.paper

import org.bukkit.plugin.java.JavaPlugin

class PaperVoiceLoader(
    private val voiceServer: PaperVoiceServer,
) : JavaPlugin() {
    init {
        voiceServer.plugin = this
    }

    override fun onEnable() {
        voiceServer.onInitialize()
    }

    override fun onDisable() {
        voiceServer.onShutdown()
    }
}
